integration/test/SetupHelpers.hs

```diff
@@ -107,3 +108,20 @@ fullSearchWithAll =
         let remoteDomains = List.delete ownDomain $ [env.domain1, env.domain2] <> env.dynamicDomains
         addFullSearchFor remoteDomains val
     }
+
+withFederatingBackendsAllowDynamic :: HasCallStack => Int -> ((String, String, String) -> App a) -> App a
+withFederatingBackendsAllowDynamic n a = do
```

Comment:
Nitpick: not sure about calling the variable here `a`, instead of `f` or something a bit more explicit.